### PR TITLE
Feature: programmatically calculates percentages in donut-chart if none are defined in data-set

### PR DESCRIPTION
**Pull request by jenjwong**
## Description
Modifies the following files:

**src/charts/donut.js**
-changes donut-chart percentage parameter from required to optional
-adds helper function _sumValues_ 
-modifies _cleanData_ function to calculate percentages if none are defined in data-set

**src/charts/helpers/common.js**
-adds function _calculatePercent_ that returns percent with customizable number of decimal places 
-documents API in JSDoc
-organizes helper functions in alphabetical order

**test/json/donutDataFivePlusOtherNoPercent.json**
-adds new donut-chart data-set with no percentage values

**test/fixtures/donutChartDataBuilder.js**
-adds _this.withFivePlusOtherNoPercent_ method to _DonutDataBuilder_ function

**test/specs/donut.spec.js**
-adds function that runs all existing tests for each donut data-set
-adds test verifying _cleanData_ function uses existing percentages if declared otherwise calculates values, and checks that correct values are generated

**test/specs/helpers.spec.js**
-adds test verifying _calculatePercent_ returns correct percentage
-adds test verifying _calculatePercent_ returns percent with specified number of decimal places

## Motivation and Context
This change adds a new feature to the donut-chart so it can programatically calculate percentages if none are defined in data-set. [Help Wanted](https://github.com/eventbrite/britecharts/issues/242) #242

## How Has This Been Tested?

Passes all tests. Adds two new tests and runs all existing donut-chart tests for new data-set.

## Screenshots (if appropriate):
percent

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/charts/helpers/common.js [hunk 1]
```diff
@@ -1,5 +1,18 @@
...
-    'use strict'
+    'use strict';
+
+    const d3Format = require('d3-format');
+
+    /**
+     * Calculates percentage of value from total
+     * @param  {Number}  value    Value to check
+     * @param  {Number}  total    Sum of values
+     * @param  {String}  decimals Specifies number of decimals https://github.com/d3/d3-format
```
Comment (Golodhros) [inline src/charts/helpers/common.js]: Thanks for the reference here!

File: test/specs/helpers.spec.js [hunk 2]
```diff
@@ -112,6 +112,10 @@ define([
             it('should return false passed a non integer', () => {
                 expect(common.isInteger(3.2)).toEqual(false);
             });
+
+            it('should calculate percent from value and total', () => {
```
Comment (Golodhros) [inline test/specs/helpers.spec.js]: Could you add a test for a different format value?

Comment (jenjwong) [inline test/specs/helpers.spec.js]: Sure!

**Comment by miglesiasEB** [comment]
This looks perfect, thanks a lot!!

You will need to run `npm run docs` in order to generate the updated docs (we are going to try to fix this detail with Travis), and this will be ready!!

**Comment by jenjwong** [comment]
Fabulous! Jumping into the codebase was a lot of fun and Britecharts is a great tool! All of your work over the past two years was definitely worth it!

**Pull request merged**